vke/src/Utils.cs

```diff
@@ -67,88 +68,61 @@ public static XmlAttributeOverrides GetXmlOverrides ()
 		/// Populate a Vector3 with values from a float array
 		/// </summary>
 		public static void FromFloatArray (ref Vector3 v, float[] floats) {
-			if (floats.Length > 0)
-				v.X = floats[0];
-			if (floats.Length > 1)
-				v.Y = floats[1];
-			if (floats.Length > 2)
-				v.Z = floats[2];
+			v = Unsafe.As<float[], Vector3[]>(ref floats)[0];
 		}
 		/// <summary>
 		/// Populate a Vector4 with values from a float array
 		/// </summary>
 		public static void FromFloatArray (ref Vector4 v, float[] floats) {
-			if (floats.Length > 0)
-				v.X = floats[0];
-			if (floats.Length > 1)
-				v.Y = floats[1];
-			if (floats.Length > 2)
-				v.Z = floats[2];
-			if (floats.Length > 3)
-				v.W = floats[3];
-        }
+			v = Unsafe.As<float[], Vector4[]>(ref floats)[0];
+		}
 		/// <summary>
 		/// Populate a Quaternion with values from a float array
 		/// </summary>
-		public static void FromFloatArray (ref Quaternion v, float[] floats) {
-			if (floats.Length > 0)
-				v.X = floats[0];
-			if (floats.Length > 1)
-				v.Y = floats[1];
-			if (floats.Length > 2)
-				v.Z = floats[2];
-			if (floats.Length > 3)
-				v.W = floats[3];
+		public static void FromFloatArray (ref Quaternion q, float[] floats) {
+			q = Unsafe.As<float[], Quaternion[]>(ref floats)[0];
 		}
 		/// <summary>
 		/// Populate a Vector2 with values from a byte array starting at offset
 		/// </summary>
 		public static void FromByteArray (ref Vector2 v, byte[] byteArray, int offset) {
-            v.X = BitConverter.ToSingle (byteArray, offset);
-            v.Y = BitConverter.ToSingle (byteArray, offset + 4);
+			v = Unsafe.As<byte[], Vector2[]>(ref Unsafe.AsRef(byteArray.SubArray(offset, 8)))[0];
```

Comment:
I don't know how large the byte arrays that typically get passed into this are but if they are typically large I would consider passing them in as a `ref`